File: CS/NetCore/net/13_Touch/TouchKeyboard.cs
using GrapeCity.Win.Editors;

namespace InputManWin13_Demo._13_Touch;

public partial class TouchKeyboard : InputManWin13_Demo.DemoBase_settings
{
    public TouchKeyboard()
    {
        InitializeComponent();
        InitializeForm();
    }

    private void InitializeForm()
    {
        // xmlファイルからデータ取得
        this.dataSet1.ReadXml(Path.Combine(Application.StartupPath, @"Resources\prefecture.xml"));
        gcComboBox1.DataSource = dataSet1.Tables[0];
        gcComboBox1.SelectedIndex = -1;

        for (int i = 1; i < gcComboBox1.ListColumns.Count; i++)
        {
            gcComboBox1.ListColumns[i].AutoWidth = true;
        }

        // xmlファイルからデータ取得（gcMaskedComboBox）
        this.dataSet2.ReadXml(Path.Combine(Application.StartupPath, @"Resources\pref.xml"));
        gcMaskedComboBox1.DataSource = dataSet2.Tables[0];
        gcMaskedComboBox1.SelectedIndex = -1;

        for (int i = 1; i < gcMaskedComboBox1.ListColumns.Count; i++)
        {
            gcMaskedComboBox1.ListColumns[i].AutoWidth = true;
        }

        gcComboBox2.SelectedIndex = 2;
        // InputScopeの設定
        gcComboBox2.SelectedIndexChanged += (s, e) => SetInputScope();
        gcComboBox3.SelectedIndex = 1;
        // AutoHideTouchKeyboardの設定
        gcComboBox3.SelectedIndexChanged += (s, e) => SetAutoHideTouchKeyboard();

        gcTextBox1.GotFocus += (s, e) => ShowTouchKeyboard();
        gcMask1.GotFocus += (s, e) => ShowTouchKeyboard();
        gcCharMask1.GotFocus += (s, e) => ShowTouchKeyboard();
        gcDateTime1.GotFocus += (s, e) => ShowTouchKeyboard();
        gcDate1.GotFocus += (s, e) => ShowTouchKeyboard();
        gcTime1.GotFocus += (s, e) => ShowTouchKeyboard();
        gcTimeSpan1.GotFocus += (s, e) => ShowTouchKeyboard();
        gcNumber1.GotFocus += (s, e) => ShowTouchKeyboard();
        gcComboBox1.GotFocus += (s, e) => ShowTouchKeyboard();
        gcMaskedComboBox1.GotFocus += (s, e) => ShowTouchKeyboard();

        gcTextBox1.LostFocus += (s, e) => HideTouchKeyboard();
        gcMask1.LostFocus += (s, e) => HideTouchKeyboard();
        gcCharMask1.LostFocus += (s, e) => HideTouchKeyboard();
        gcDateTime1.LostFocus += (s, e) => HideTouchKeyboard();
        gcDate1.LostFocus += (s, e) => HideTouchKeyboard();
        gcTime1.LostFocus += (s, e) => HideTouchKeyboard();
        gcTimeSpan1.LostFocus += (s, e) => HideTouchKeyboard();
        gcNumber1.LostFocus += (s, e) => HideTouchKeyboard();
        gcComboBox1.LostFocus += (s, e) => HideTouchKeyboard();
        gcMaskedComboBox1.LostFocus += (s, e) => HideTouchKeyboard();
    }

    private void ShowTouchKeyboard()
    {
        if (checkBox1.Checked)
        {
            // タッチキーボードを表示
            gcIme1.ShowTouchKeyboard();
        }
    }

    private void HideTouchKeyboard()
    {
        if (checkBox2.Checked)
        {
            // タッチキーボードを表示
            gcIme1.HideTouchKeyboard();
        }
    }

    private void SetInputScope()
    {
        InputScopeNameValue inputScope = (InputScopeNameValue)Enum.Parse(typeof(InputScopeNameValue), gcComboBox2.SelectedItem.Text);
        gcIme1.SetInputScope(gcTextBox1, inputScope);
        gcIme1.SetInputScope(gcMask1, inputScope);
        gcIme1.SetInputScope(gcCharMask1, inputScope);
        gcIme1.SetInputScope(gcDateTime1, inputScope);
        gcIme1.SetInputScope(gcDate1, inputScope);
        gcIme1.SetInputScope(gcTime1, inputScope);
        gcIme1.SetInputScope(gcTimeSpan1, inputScope);
        gcIme1.SetInputScope(gcNumber1, inputScope);
        gcIme1.SetInputScope(gcComboBox1, inputScope);
        gcIme1.SetInputScope(gcMaskedComboBox1, inputScope);
    }

    private void SetAutoHideTouchKeyboard()
    {
        gcTextBox1.DropDown.AutoHideTouchKeyboard = (AutoHideTouchKeyboard)gcComboBox3.SelectedIndex;
        gcDate1.DropDown.AutoHideTouchKeyboard = (AutoHideTouchKeyboard)gcComboBox3.SelectedIndex;
        gcDateTime1.DropDown.AutoHideTouchKeyboard = (AutoHideTouchKeyboard)gcComboBox3.SelectedIndex;
        gcTime1.DropDown.AutoHideTouchKeyboard = (AutoHideTouchKeyboard)gcComboBox3.SelectedIndex;
        gcNumber1.DropDown.AutoHideTouchKeyboard = (AutoHideTouchKeyboard)gcComboBox3.SelectedIndex;
        gcComboBox1.DropDown.AutoHideTouchKeyboard = (AutoHideTouchKeyboard)gcComboBox3.SelectedIndex;
        gcMaskedComboBox1.DropDown.AutoHideTouchKeyboard = (AutoHideTouchKeyboard)gcComboBox3.SelectedIndex;
    }
}

File: VB/NetCore/net/13_Touch/TouchKeyboard.vb
Imports System.Windows.Forms
Imports GrapeCity.Win.Editors

Namespace _13_Touch
    Partial Public Class TouchKeyboard
        Inherits InputManWin13_Demo.DemoBase_settings
        Public Sub New()
            InitializeComponent()
            InitializeForm()
        End Sub

        Private Sub InitializeForm()
           ' xmlファイルからデータ取得
            Dim filePath As System.String = Application.StartupPath
            Me.dataSet1.ReadXml(filePath & "\Resources\prefecture.xml")
            gcComboBox1.DataSource = dataSet1.Tables(0)
            gcComboBox1.SelectedIndex = -1

            For i As Integer = 1 To gcComboBox1.ListColumns.Count - 1
                gcComboBox1.ListColumns(i).AutoWidth = True
            Next

            ' xmlファイルからデータ取得(GcMaskedComboBox)
            Me.DataSet2.ReadXml(filePath & "\Resources\pref.xml")
            GcMaskedComboBox1.DataSource = DataSet2.Tables(0)
            GcMaskedComboBox1.SelectedIndex = -1

            For i As Integer = 1 To GcMaskedComboBox1.ListColumns.Count - 1
                GcMaskedComboBox1.ListColumns(i).AutoWidth = True
            Next

            gcComboBox2.SelectedIndex = 2
            gcComboBox3.SelectedIndex = 1

        End Sub

        Private Sub Control_GotFocus(ByVal sender As Object, ByVal e As EventArgs) Handles gcTextBox1.GotFocus, gcMask1.GotFocus, gcCharMask1.GotFocus, gcDateTime1.GotFocus, gcDate1.GotFocus, gcTime1.GotFocus, gcTimeSpan1.LostFocus, gcNumber1.GotFocus, gcComboBox1.GotFocus, GcMaskedComboBox1.GotFocus
            If checkBox1.Checked Then
                ' タッチキーボードを表示
                gcIme1.ShowTouchKeyboard()
            End If
        End Sub

        Private Sub Control_LostFocus(ByVal sender As Object, ByVal e As EventArgs) Handles gcTextBox1.LostFocus, gcMask1.LostFocus, gcCharMask1.LostFocus, gcDateTime1.LostFocus, gcDate1.LostFocus, gcTime1.LostFocus, gcTimeSpan1.LostFocus, gcNumber1.LostFocus, gcComboBox1.LostFocus, GcMaskedComboBox1.LostFocus
            If checkBox2.Checked Then
                ' タッチキーボードを非表示
                gcIme1.HideTouchKeyboard()
            End If
        End Sub

        Private Sub gcComboBox2_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles gcComboBox2.SelectedIndexChanged
            ' InputScopeの設定
            Dim inputScope As InputScopeNameValue = DirectCast([Enum].Parse(GetType(InputScopeNameValue), gcComboBox2.SelectedItem.Text), InputScopeNameValue)
            gcIme1.SetInputScope(gcTextBox1, inputScope)
            gcIme1.SetInputScope(gcMask1, inputScope)
            gcIme1.SetInputScope(gcCharMask1, inputScope)
            gcIme1.SetInputScope(gcDateTime1, inputScope)
            gcIme1.SetInputScope(gcDate1, inputScope)
            gcIme1.SetInputScope(gcTime1, inputScope)
            gcIme1.SetInputScope(gcTimeSpan1, inputScope)
            gcIme1.SetInputScope(gcNumber1, inputScope)
            gcIme1.SetInputScope(gcComboBox1, inputScope)
            gcIme1.SetInputScope(GcMaskedComboBox1, inputScope)
        End Sub

        Private Sub gcComboBox3_SelectedIndexChanged(sender As Object, e As EventArgs) Handles gcComboBox3.SelectedIndexChanged
            ' AutoHideTouchKeyboardの設定
            gcTextBox1.DropDown.AutoHideTouchKeyboard = DirectCast(gcComboBox3.SelectedIndex, AutoHideTouchKeyboard)
            gcDate1.DropDown.AutoHideTouchKeyboard = DirectCast(gcComboBox3.SelectedIndex, AutoHideTouchKeyboard)
            gcDateTime1.DropDown.AutoHideTouchKeyboard = DirectCast(gcComboBox3.SelectedIndex, AutoHideTouchKeyboard)
            gcTime1.DropDown.AutoHideTouchKeyboard = DirectCast(gcComboBox3.SelectedIndex, AutoHideTouchKeyboard)
            gcNumber1.DropDown.AutoHideTouchKeyboard = DirectCast(gcComboBox3.SelectedIndex, AutoHideTouchKeyboard)
            gcComboBox1.DropDown.AutoHideTouchKeyboard = DirectCast(gcComboBox3.SelectedIndex, AutoHideTouchKeyboard)
            GcMaskedComboBox1.DropDown.AutoHideTouchKeyboard = DirectCast(gcComboBox3.SelectedIndex, AutoHideTouchKeyboard)

        End Sub
    End Class
End Namespace